ngx_http_proxy_connect_module.c

```diff
@@ -1366,17 +1396,7 @@ ngx_http_proxy_connect_handler(ngx_http_request_t *r)
         return NGX_HTTP_FORBIDDEN;
     }
 
-    ctx = ngx_pcalloc(r->pool, sizeof(ngx_http_proxy_connect_ctx_t));
-    if (ctx == NULL) {
-        return NGX_ERROR;
-    }
-
-    ctx->buf.pos = (u_char *) NGX_HTTP_PROXY_CONNECT_ESTABLISTHED;
-    ctx->buf.last = ctx->buf.pos +
-                    sizeof(NGX_HTTP_PROXY_CONNECT_ESTABLISTHED) - 1;
-    ctx->buf.memory = 1;
-
-    ngx_http_set_ctx(r, ctx, ngx_http_proxy_connect_module);
+    ctx = ngx_http_get_module_ctx(r, ngx_http_proxy_connect_module);;
```

Comment:
Move code logic to ngx_http_proxy_connect_post_read_handler. ( allocate & initialize `ctx` )